### PR TITLE
-buildmode=pie is not supported for the mips arch

### PR DESCRIPTION
**Pull request by XiaodongLoong**
reference:
https://github.com/docker/cli/pull/2507
https://github.com/containerd/containerd/commit/4c99c81326f4026fb8c0b8c5e10542205d99c321

Signed-off-by: Xiaodong Liu <liuxiaodong@loongson.cn>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
Can we split out the common part into a separate script?

**Comment by XiaodongLoong** [comment]
> Can we split out the common part into a separate script?

Very Good advice!  I will try my best to make it!

**Comment by XiaodongLoong** [comment]
@tianon I have modified the commit according to comment from AkihiroSuda , please review, Thanks!

File: hack/dockerfile/install/rootlesskit.installer [hunk 1]
```diff
@@ -21,7 +21,7 @@ install_rootlesskit() {
 
 install_rootlesskit_dynamic() {
 	export ROOTLESSKIT_LDFLAGS="-linkmode=external" install_rootlesskit
-	export BUILD_MODE="-buildmode=pie"
+	export BUILD_MODE=${GO_BUILDMODE}
```
Comment (AkihiroSuda) [inline hack/dockerfile/install/rootlesskit.installer:24]: Can we use `BUILD_MODE` everywhere instead of introducing a new variable `GO_BUILDMODE`

Comment (XiaodongLoong) [reply hack/dockerfile/install/rootlesskit.installer:24]: I think `GO_BUILDMODE` is more specific and accurate. That change `GO_BUILDMODE` to `BUILD_MODE`  is also OK.

Comment (tianon) [reply hack/dockerfile/install/rootlesskit.installer:24]: Yeah, IMO this should go the other way and update everything to use the more specific `GO_BUILDMODE` name.

**Review by tianon — APPROVED** [review]
LGTM

**Review by AkihiroSuda — APPROVED** [review]

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by XiaodongLoong** [comment]
@AkihiroSuda I use the conflict resolver provided by Github, is it OK？If not, I can close and reopen PR, rebase and subbmit a new commit againe.

**Comment by AkihiroSuda** [comment]
Please squash

**Comment by tianon** [comment]
LGTM :+1:

**Comment by XiaodongLoong** [comment]
please merge, thanks!

**Comment by XiaodongLoong** [comment]
@AkihiroSuda @tianon @thaJeztah

**Comment by XiaodongLoong** [comment]
@AkihiroSuda I have squashed the commits, Thanks!

**Review by AkihiroSuda — APPROVED** [review]

**Comment by XiaodongLoong** [comment]
@tianon I rebased the commit according to AkihiroSuda, please review, I am so sorry for you have to review it again.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**